lib/api/data_api.dart

```diff
@@ -123,9 +123,9 @@ class DataApi {
 
       var electiveSubjectDetails = <SubjectDetails>[];
 
-      if (response.data[1] != null ||
-          response.data[1]['subject'] != null ||
-          response.data[1] != []) {
+      if (
+        response.data.length > 2 && 

```

Comment:
Can you check if the app works fine after doing the change below?

```suggestion
        response.data.length >= 2 && 
```

Reply:
Yes the app is working fine, thanks for consideration the bug.